server/cron/calendar_cron.go

```diff
@@ -482,6 +510,50 @@ func processFailingHostExistingCalendarEvent(
 	return nil
 }
 
+func getBodyTag(ctx context.Context, ds fleet.Datastore, host fleet.HostPolicyMembershipData, policyIDtoPolicy *sync.Map,
+	logger kitlog.Logger) string {
+	var updatedBodyTag string
+	policyIDs := strings.Split(host.FailingPolicyIDs, ",")
+	if len(policyIDs) == 1 && policyIDs[0] != "" {
+		var policy *calendar.PolicyLiteWithMeta
+		policyAny, ok := policyIDtoPolicy.Load(policyIDs[0])
+		if !ok {
+			id, err := strconv.ParseUint(policyIDs[0], 10, 64)
+			if err != nil {
+				level.Error(logger).Log("msg", "parse policy id", "err", err)
+				// Do nothing
+				return ""
+			}
+			policyLite, err := ds.PolicyLite(ctx, uint(id))
+			if err != nil {
+				level.Error(logger).Log("msg", "get policy", "err", err)
+				// Do nothing
+				return ""
+			}
+			policy = new(calendar.PolicyLiteWithMeta)
+			policy.PolicyLite = *policyLite
+			policyIDtoPolicy.Store(policyIDs[0], policy)
```

Comment:
how would this scenario (policy id ok, get policy ok, but mapping wasn't in `policyIDtoPolicy`, if I'm understanding right?) occur?

Reply:
I'm curious to understand better where/how the `policyIDtoPolicy` cache is intitialized/persisted

Reply:
The `policyIDtoPolicy` map is stored in memory and starts out empty. It is only initialized/used during a cron run. So when a policy is loaded from DB, we put it into this map. The next time we need policy description, we check the map it we already have it.